superset/assets/visualizations/deckgl/DeckGLContainer.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import MapGL from 'react-map-gl';
+import DeckGL from 'deck.gl';
+
+const propTypes = {
+  viewport: PropTypes.object.isRequired,
+  layers: PropTypes.array.isRequired,
+  setControlValue: PropTypes.func.isRequired,
+  mapStyle: PropTypes.string,
+  mapboxApiAccessToken: PropTypes.string.isRequired,
+  onViewportChange: PropTypes.func,
+};
+const defaultProps = {
+  mapStyle: 'light',
+  onViewportChange: () => {},
+};
+
+export default class DeckGLContainer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      viewport: props.viewport,
+    };
+    this.tick = this.tick.bind(this);
+  }
+  componentWillMount() {
+    const timer = setInterval(this.tick, 1000);
+    this.setState({ timer });
+  }
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      viewport: {
+        ...nextProps.viewport,
+      },
+    });
+  }
+  componentWillUnmount() {
+    this.clearInterval(this.state.timer);
+  }
+  onViewportChange(viewport) {
+    const vp = Object.assign({}, viewport);
```

Comment:
nit `{ ...spread }`, need a lint rule for this.

Reply:
What about we do the spread change as a large linting PR across the board that would clean all Object.assign and set up the linting rule?

Reply:
https://github.com/apache/incubator-superset/issues/3835